internal/fdo.go

```diff
@@ -232,23 +297,33 @@ func newFdoIconData(theme string, size int, desktopPath string) *fdoIconData {
 
 	scanner := bufio.NewScanner(file)
 	fdoApp := fdoIconData{name: "", iconName: "", iconPath: "", exec: ""}
+	var currentSection string
 	for scanner.Scan() {
 		line := scanner.Text()
-		if strings.HasPrefix(line, "Name=") {
-			name := strings.SplitAfter(line, "=")
-			fdoApp.name = name[1]
-		} else if strings.HasPrefix(line, "Icon=") {
-			icon := strings.SplitAfter(line, "=")
-			fdoApp.iconName = icon[1]
-			fdoApp.iconPath = fdoLookupIconPath(theme, size, fdoApp.iconName)
-			if fdoApp.iconPath == "" {
-				fyne.LogError("Could not find path for icon "+fdoApp.iconName, nil)
+		if strings.HasPrefix(line, "[") {
+			currentSection = line
+		}
+		if currentSection == "[Desktop Entry]" {
```

Comment:
invert the logic here again and you can avoid this additional indentation of the main logic